File: dat_learning_system-fe/src/types_interfaces/roadmap.ts
export interface RoadmapStep {
    id: number;
    title: string;
    nodeType: 'EBook' | 'Course' | 'Instruction';
    content?: string;
    linkedResourceId?: string;
    linkedResourceTitle?: string;
    linkedResourceDescription?: string;
    sortOrder: number;
}

export interface RoadmapResponse {
    id: number;
    title: string;
    description: string;
    targetRole?: string;
    stepCount: number;
    steps: RoadmapStep[];
}

export interface RoadmapRequest {
    title: string;
    description: string;
    targetRole?: string;
    linkedResourceId?: string | null;
}

export interface RoadmapGlobalSourceDto {
    value: string;       // e.g., "EBook - 1" or "Course - uuid"
    title: string;       // The Name of the book/course
    description: string; // Brief summary
    type: string;        // "EBook" or "Course"
}

File: dat_learning_system-be/LMS.Backend/Data/Entities/Roadmap.cs
namespace LMS.Backend.Data.Entities;

public class RoadMap
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? TargetRole { get; set; } 
    public string? ThumbnailUrl { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<RoadmapStep> Nodes { get; set; } = new List<RoadmapStep>();
}

public class RoadmapStep
{
    public int Id { get; set; }
    public int LearningPathId { get; set; }
    public RoadMap RoadMap { get; set; } = null!;

    public string Title { get; set; } = string.Empty;
    public string NodeType { get; set; } = "Instruction"; // EBook, Course, Instruction
    
    // This is the "What to do" or "How to do it"
    public string? Content { get; set; } 
    
    // If NodeType is EBook, store the ID here to link to your existing Library
    public int? LinkedResourceId { get; set; } 

    public int SortOrder { get; set; }
}